pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
Now that you say I agree a lot of those timing metrics are quite useless (not sure anyone use that http endpoint). The metrics publisher gets counters that reset (so mean, count, min/max etc are only for the current flush interval), I made the http endpoint just as after thought, but your right without a flush interval (or a sum as you suggest) they are quite useless. 

Looking at the prometheus histogram, looks better than the histogram that I copied from the go metrics lib.